src/duke/Parser.java

```diff
@@ -0,0 +1,337 @@
+package duke;
+
+import java.io.IOException;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * This is an Action class that obtains a sentence as input that
+ * can be deciphered to create tasks in the duke.Duke system
+ *
+ * @author  Hsiao Jiet
+ * @version 1.0
+ * @since   2022-1-15
+ */
+
+public class Parser {
+    protected String userInput;
+    protected TaskList tasks;
+    protected String[] listOfUserInput;
+    protected Storage storage;
+    String line = "\n____________________________________________________________\n";
+
+    enum Commands {
+        todo, deadline, event, list, mark, unmark, delete, find, bye
+    }
+
+    public Parser(String input, TaskList list, Storage storage) {
+        userInput = input;
+        tasks = list;
+        this.storage = storage;
+    }
+
+    /**
+     * First split of user input to obtain Action word from other user input words
+     */
+    public void splitUserInput() {
+        listOfUserInput = userInput.split(" ");
+    }
+
+    /**
+     * Obtains the Action word from user input
+     */
+    public String obtainCommandWord() {
+        return listOfUserInput[0];
+    }
+
+    /**
+     * Obtains title. Applicable for creating Todo, duke.Deadline and duke.Event
+     */
+    public String obtainTitle() throws DukeException {
```

Comment:
I think it will help greatly if this method was named more descriptively, in particular, what is a `Title`? What do you think?